### DISCUSSION TITLE
Listening for text inside of an action event listener.

### DISCUSSION THREAD

**Discussion by kapisolec** [telegraf/telegraf#1722] [Q&A] [answered]
Hi, I would like to know if there is a way to "chain" listeners in another listener, for instance:

```javascript
bot.action("Add data", async (ctx) => {
  const { reply_markup } = Keyboard.inline([
    ["❌Cancel"], // 1st row
  ]);

  const options = {
    reply_markup,
    parse_mode: "HTML",
  };

  await ctx.reply("Please enter the data you want to add", options);

  // listen for changes in this context
  bot.on("text", async (ctx) => {
    console.log(ctx);
    // add data...
  });
})
```

And if not, is there a way to "save" the messageId or the call itself for the current user from this `bot.action` listener that way so I know that the data they pass will be the data for that exact `bot.action`

**Comment by MKRhere** [comment, marked as answer]
You shouldn't add a `bot.on` inside another listener - this will cause it to keep adding a new listener every time and run out of memory.

Instead, use scenes. Here's an [example](https://github.com/feathers-studio/telegraf-docs/blob/4feffcd893a86ed1fcc761279d17328beb4a2f21/examples/scenes/scenes.ts). You can also check out [this answer](https://github.com/telegraf/telegraf/issues/705#issuecomment-549056045) for more details.
